homeassistant/components/minecraft_server/config_flow.py

```diff
@@ -0,0 +1,103 @@
+"""Config flow for Minecraft Server integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+
+from . import MinecraftServer
+from .const import (  # pylint: disable=unused-import
+    CONF_UPDATE_INTERVAL,
+    DEFAULT_HOST,
+    DEFAULT_NAME,
+    DEFAULT_PORT,
+    DEFAULT_UPDATE_INTERVAL_SECONDS,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class MinecraftServerConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Minecraft Server."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize Minecraft Server config flow."""
+        pass
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            # No configuration data available yet, show default configuration form.
+            return await self._show_config_form()
+
+        # User inputs.
+        name = user_input[CONF_NAME]
+        host = user_input[CONF_HOST]
+        port = user_input[CONF_PORT]
+        update_interval = user_input[CONF_UPDATE_INTERVAL]
+
+        # Abort in case the host was already configured before.
+        await self.async_set_unique_id(f"{host.lower()}-{port}")
+        self._abort_if_unique_id_configured()
+
+        errors = {}
+
+        # Validate port configuration (limit to user and dynamic port range).
+        if (port < 1024) or (port > 65535):
+            errors["base"] = "invalid_port"
+        # Validate update interval configuration (min: 5s, max: 24h).
+        elif (update_interval < 5) or (update_interval > 86400):
```

Comment:
Theoratically yes, it's possible with a voluptuous schema. But if I do it that way, the config flow UI shows a slider instead of a number input, which I was not able to get rid of. If you know the solution I would be happy to move the validation to the schema.

Reply:
Limiting, because ports have different ranges (system, registered and dynamic). My integration doesn't allow system ports to be used. Refer to [List of TCP and UDP port numbers](https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers).

Reply:
Nice to me